force-app/main/default/classes/MonthlyIterator.cls

```diff
@@ -0,0 +1,67 @@
+public with sharing class MonthlyIterator implements RecurrenceIteratorInterface {
+    private List<DateTime> result;
+    private Date runningDate;
+    @TestVisible
+    private RecurrenceRule rRule;
+
+    public MonthlyIterator(RecurrenceRule rRule) {
+        this.rRule = rRule;
+    }
+
+    public List<DateTime> iterate(DateTime dateTimeStart) {
+        result = new List<DateTime>();
+        runningDate = dateTimeStart.date();
+        Integer startDay;
+
+        do {
+            if (rRule.getBySetPos() != null && rRule.getByMonthDays() != null) {
+                //The startDay variable will hold the max value that is returned by rRule.getByMonthDays.
+                startDay = getMaxByMonthDays(rRule.getByMonthDays());
+                result.add(DateTime.newInstance(runningDate, dateTimeStart.time()));
+
+                //Here we are adding days after adding months. So the number of days we are adding will
+                //be the difference of startDay and runningDate.day().
+
+                //Eg : if startdate is 1/31/2020 and we are creating occurences until 04/30/2020. The start date is added
```

Comment:
Super smart, there is still something bugging me at the back of my mind. Can you add a test that starts on 12/31 and goes every other month?

Reply:
@jjbennett , There is a test that tests this scenario. The name of the test method is `shouldIterateEveryLastDayOfTheMonthUntilGivenEndDateOverLeapYearWithIntervalOf2`. Let me know if you want me to add additional tests